v0.15 build failure with Apple Clang

Attempting to upgrade Homebrew's IWYU returned CI build failures here: https://github.com/Homebrew/homebrew-core/pull/65850

Homebrew builds with the system compiler (clang). This returned a long series of errors that look like
```
In file included from /usr/local/opt/llvm/include/llvm/ADT/SmallVector.h:17:
/usr/local/opt/llvm/include/llvm/Support/AlignOf.h:49:3: error: C++ requires a type specifier for all declarations
  alignas(::llvm::detail::AlignerImpl<T, Ts...>) char buffer[sizeof(
  ^
/usr/local/opt/llvm/include/llvm/Support/AlignOf.h:49:49: error: expected ';' at end of declaration list
  alignas(::llvm::detail::AlignerImpl<T, Ts...>) char buffer[sizeof(
                                                ^
                                                ;
In file included from /tmp/include-what-you-use-20201128-99314-l5cdfv/iwyu_ast_util.cc:12:
In file included from /tmp/include-what-you-use-20201128-99314-l5cdfv/iwyu_ast_util.h:19:
In file included from /tmp/include-what-you-use-20201128-99314-l5cdfv/iwyu_port.h:16:
In file included from /usr/local/opt/llvm/include/llvm/Support/raw_ostream.h:16:
In file included from /usr/local/opt/llvm/include/llvm/ADT/SmallVector.h:17:
/usr/local/opt/llvm/include/llvm/Support/AlignOf.h:49:3: error: C++ requires a type specifier for all declarations
  alignas(::llvm::detail::AlignerImpl<T, Ts...>) char buffer[sizeof(
  ^
/usr/local/opt/llvm/include/llvm/Support/AlignOf.h:49:49: error: expected ';' at end of declaration list
  alignas(::llvm::detail::AlignerImpl<T, Ts...>) char buffer[sizeof(
                                                ^
                                                ;
```

Compiling IWYU with LLVM clang instead is successful.

Build logs:
[iwyu-log.tar.gz](https://github.com/include-what-you-use/include-what-you-use/files/5685157/iwyu-log.tar.gz)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38,30 +38,6 @@ include_directories(
   ${CLANG_INCLUDE_DIRS}
   )
 
-set(LLVM_LINK_COMPONENTS
-  Option
-  Support
-  X86AsmParser
-  X86Desc
-  X86Info
-  )
-
-add_llvm_executable(include-what-you-use
-  iwyu.cc
-  iwyu_ast_util.cc
-  iwyu_cache.cc
-  iwyu_driver.cc
-  iwyu_getopt.cc
-  iwyu_globals.cc
-  iwyu_include_picker.cc
-  iwyu_lexer_utils.cc
-  iwyu_location_util.cc
-  iwyu_output.cc
-  iwyu_path_util.cc
-  iwyu_preprocessor.cc
-  iwyu_verrs.cc
-  )
-
 # Synthesize a clang-resource-headers target for out-of-tree builds (in-tree
 # already has it available by default)
 if (NOT IWYU_IN_TREE)
@@ -88,6 +64,30 @@ if (NOT IWYU_IN_TREE)
 
   add_custom_target(clang-resource-headers ALL DEPENDS ${out_files})
 endif()
+
+set(LLVM_LINK_COMPONENTS
+  Option
+  Support
+  X86AsmParser
+  X86Desc
+  X86Info
+  )
+
+add_llvm_executable(include-what-you-use
+  iwyu.cc
+  iwyu_ast_util.cc
+  iwyu_cache.cc
+  iwyu_driver.cc
+  iwyu_getopt.cc
+  iwyu_globals.cc
+  iwyu_include_picker.cc
+  iwyu_lexer_utils.cc
+  iwyu_location_util.cc
+  iwyu_output.cc
+  iwyu_path_util.cc
+  iwyu_preprocessor.cc
+  iwyu_verrs.cc
+  )
 
 # Add a dependency on clang-resource-headers to ensure the builtin headers are
 # available when IWYU is executed from the build dir.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -93,6 +93,11 @@ add_llvm_executable(include-what-you-use
 # available when IWYU is executed from the build dir.
 add_dependencies(include-what-you-use clang-resource-headers)
 
+# LLVM requires C++14, so follow suit.
+set_property(TARGET include-what-you-use PROPERTY CXX_STANDARD_REQUIRED ON)
+set_property(TARGET include-what-you-use PROPERTY CXX_STANDARD 14)
+set_property(TARGET include-what-you-use PROPERTY CXX_EXTENSIONS OFF)
+
 if (MINGW)
   # Work around 'too many sections' error with MINGW/GCC
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wa,-mbig-obj")

